libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -692,7 +692,8 @@ void _serializeConnect( const IotMqttConnectInfo_t * pConnectInfo,
     }
 
     /* Username and password depend on MQTT mode. */
-    if( pConnectInfo->awsIotMqttMode == true )
+    if( ( pConnectInfo->pUserName == NULL ) && 
+        ( pConnectInfo->awsIotMqttMode == true ) )
     {
         /* Set the username flag for AWS IoT metrics. The AWS IoT MQTT server
```

Comment:
If `pConnectInfo->awsIotMqttMode == true` and `pConnectInfo->pUserName != NULL` and `AWS_IOT_MQTT_ENABLE_METRICS == 1` is defined, we should log an error since the metrics will not be able to use the username area.

Reply:
This limitation got corrected based on offline conversation. I've updated the code.